src/engagement_db_to_analysis/engagement_db_to_analysis.py

```diff
@@ -112,26 +150,74 @@ def generate_analysis_files(user, google_cloud_credentials_file_path, pipeline_c
 
     run_automated_analysis(messages_by_column, participants_by_column, pipeline_config.analysis, f"{output_dir}/automated-analysis")
 
+
+def generate_analysis_files(user, google_cloud_credentials_file_path, pipeline_config, uuid_table, engagement_db, rapid_pro,
+                            membership_group_dir_path, output_dir, cache_path=None, dry_run=False):
+    """
+    :type pipeline_config: src.pipeline_configuration_spec.PipelineConfiguration
+    """
+
+    analysis_dataset_configurations = pipeline_config.analysis.dataset_configurations
+    # TODO: Tidy up which functions get passed analysis_configs and which get passed dataset_configurations
+
+    if cache_path is None:
+        cache = None
+        log.warning(f"No `cache_path` provided. This tool will perform a full download of project messages from engagement database")
+    else:
+        log.info(f"Initialising EngagementAnalysisCache at '{cache_path}/engagement_db_to_analysis'")
+        cache = AnalysisCache(f"{cache_path}/engagement_db_to_analysis")
+
+    engagement_db_datasets = []
+    for config in analysis_dataset_configurations:
+        engagement_db_datasets.extend(config.engagement_db_datasets)
+    messages_map = get_messages_in_datasets(engagement_db, engagement_db_datasets, cache, dry_run)
+
+    messages = [message for messages_list in messages_map.values() for message in messages_list]
+    messages_traced_data = _convert_messages_to_traced_data(user, messages)
+
+    messages_traced_data = filter_messages(user, messages_traced_data, pipeline_config)
+
+    consent_withdrawn_uuids = get_consent_withdrawn_participant_uuids(user, pipeline_config, messages_traced_data)
+
+    messages_traced_data_clone = messages_traced_data.copy()
+    messages_by_column, participants_by_column = process_data(
+        user, google_cloud_credentials_file_path, pipeline_config, analysis_dataset_configurations,
+        membership_group_dir_path, messages_traced_data_clone, consent_withdrawn_uuids
+    )
+    export_analysis_files(pipeline_config, messages_by_column, participants_by_column, f"{output_dir}/{MAIN_ANALYSIS_DIR}")
+    
+    channel_operators = get_channel_operators(messages)
+    for channel_operator in channel_operators:
```

Comment:
Do we need to run for each individual channel all the time (which will add quite a bit to pipeline run time and storage requirements), or can we require channels to be individually specified using the `channel_group_analysis` configuration?

Reply:
[This](https://github.com/AfricasVoices/Engagement-Data-Pipeline/pull/279#discussion_r1361922245) was the initial reason for not including it. I think we should run for each individual channel all the time. since maps are taking a lot of storage we can use the filters created earlier to reduce the regions being analysed.